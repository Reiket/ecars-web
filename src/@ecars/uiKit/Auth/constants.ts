import {withBlockClass} from 'ecars-web-lib';
import {AuthPageWrapper} from '@ecars/uiKit/Auth/components/AuthPageWrapper';

export const AuthPageWrapperHOC = withBlockClass(AuthPageWrapper, 'auth');