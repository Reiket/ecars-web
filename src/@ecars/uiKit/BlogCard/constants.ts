import type {CategoriesListType} from 'ecars-web-lib';

export const BLOG_CARD_DIRECTION = Object.freeze({
  ROW: 'row',
  COLUMN: 'column',
});

export type BlogCardDirectionType = (typeof BLOG_CARD_DIRECTION)[keyof typeof BLOG_CARD_DIRECTION];

export interface BlogCardContentProps {
  title: string;
  category: CategoriesListType;
  description?: string;
}

export interface BlogCardImageProps {
  imageUrl: string;
  alt?: string;
}

export const BLOG_CARD_TEST_ID = 'blogCardTestId';
