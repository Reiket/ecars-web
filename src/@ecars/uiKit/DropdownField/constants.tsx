import type {DropdownOption} from 'ecars-web-lib';

interface TopbarDropdownsConfig {
  label: string;
  options: DropdownOption[];
}

const shipToConfig: DropdownOption[] = [
  {
    value: 'ca',
    name: 'Canada',
    label: (
      <img
        src="./assets/images/flags/canada.svg"
        alt="ca"
      />
    ),
  },
  {
    value: 'us',
    name: 'USA',
    label: (
      <img
        src="./assets/images/flags/usa.svg"
        alt="usa"
      />
    ),
  },
  {
    value: 'eu',
    name: 'Europe',
    label: (
      <img
        src="./assets/images/flags/europe.svg"
        alt="eu"
      />
    ),
  },
];

const currencyConfig: DropdownOption[] = [
  {
    value: 'usd',
    name: 'US Dollar',
  },
  {
    value: 'eur',
    name: 'Euro',
  },
  {
    value: 'gbp',
    name: 'British Pound',
  },
];

export const topbarDropdownsConfig: TopbarDropdownsConfig[] = [
  {
    label: 'Ship to:',
    options: shipToConfig,
  },
  {
    label: 'Currency:',
    options: currencyConfig,
  },
];
