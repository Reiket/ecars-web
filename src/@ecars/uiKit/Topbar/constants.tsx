import {TopbarWrapper} from '@ecars/uiKit/Topbar/components/TopbarWrapper';
import {withBlockClass} from 'ecars-web-lib';

export const TopbarWrapperHOC = withBlockClass(TopbarWrapper, 'topbar');
export const TOPBAR_TEST_ID = 'topbarTestId';
