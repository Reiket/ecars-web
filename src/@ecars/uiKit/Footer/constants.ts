import {withBlockClass} from 'ecars-web-lib';
import {FooterWrapper} from '@ecars/uiKit/Footer/components/FooterWrapper';
import {socialLinks} from '@ecars/uiKit/SocialList/constants';
import {PageUrls} from '@ecars/constants/page-urls';
import {FooterCopyWrapper} from '@ecars/uiKit/Footer/components/FooterCopyWrapper';

export const FooterWrapperHOC = withBlockClass(FooterWrapper, 'footer');
export const FooterCopyWrapperHOC = withBlockClass(FooterCopyWrapper, 'copy');
export const footerSocialLinks = socialLinks.filter((item) => !('label' in item));
export const footerContactsLinks = socialLinks.filter((item) => 'label' in item);

export const copyLinks = [
  {
    label: 'Privacy Policy',
    href: PageUrls.PRIVACY,
  },
  {
    label: 'Terms & Conditions',
    href: PageUrls.TERMS,
  },
];

export interface FooterLink {
  label: string;
  to: string;
}

export const companyLinks: FooterLink[] = [
  {label: 'About Us', to: PageUrls.ABOUT},
  {label: 'Blog', to: PageUrls.Blog},
  {label: 'FAQ', to: PageUrls.FAQ},
];

export const carsLinks: FooterLink[] = [
  {label: 'Special offers', to: '/'},
  {label: 'New cars', to: PageUrls.NEWCARS},
  {label: 'Used cars', to: PageUrls.USEDCARS},
  {label: 'Brands', to: '/'},
];
