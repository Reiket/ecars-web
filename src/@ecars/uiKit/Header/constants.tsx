import {Icons, withBlockClass} from 'ecars-web-lib';
import {HeaderWrapper} from '@ecars/uiKit/Header/components/HeaderWrapper';
import type {PageUrlType} from '@ecars/constants/page-urls';
import {PageUrls} from '@ecars/constants/page-urls';
import {HeaderNavbarWrapper} from '@ecars/uiKit/Header/components/HeaderNavbarWrapper';
import type {ReactNode} from 'react';

export const HeaderWrapperHOC = withBlockClass(HeaderWrapper, 'header');
export const HeaderNavbarWrapperHOC = withBlockClass(HeaderNavbarWrapper, 'navbar');

interface NavbarLink {
  to: PageUrlType;
  label: string;
  isUtil?: boolean;
}

interface HeaderUtil {
  to: PageUrlType;
  icon: ReactNode;
}

export const navbarLinksConfig: NavbarLink[] = [
  {
    to: PageUrls.CATALOG,
    label: 'All Cars',
  },
  {
    to: PageUrls.ABOUT,
    label: 'About Us',
  },
  {
    to: PageUrls.Blog,
    label: 'Blog',
  },
  {
    to: PageUrls.FAQ,
    label: 'FAQ',
  },
  {
    to: PageUrls.FAVORITES,
    label: 'Favorites',
    isUtil: true,
  },
  {
    to: PageUrls.LOGIN,
    label: 'Login',
    isUtil: true,
  },
];

export const headerUtilsConfig: HeaderUtil[] = [
  {to: PageUrls.FAVORITES, icon: <Icons.FavoriteSolid />},
  {to: PageUrls.LOGIN, icon: <Icons.Account />},
];

export const headerTestScenarios = [
  {
    desc: 'initial state',
    scrolls: [0],
    expected: {sticky: false, showHeader: true},
  },
  {
    desc: 'scrolled down below topbarHeight → sticky true, showHeader false',
    scrolls: [100],
    expected: {sticky: true, showHeader: false},
  },
  {
    desc: 'scroll up again → sticky true, showHeader true',
    scrolls: [150, 50],
    expected: {sticky: true, showHeader: true},
  },
];
