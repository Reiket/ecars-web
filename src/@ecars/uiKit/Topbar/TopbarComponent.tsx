import type {FC} from 'react';
import {Topbar} from '@ecars/uiKit/Topbar';
import {TOPBAR_TEST_ID} from '@ecars/uiKit/Topbar/constants';
import {SocialList} from '@ecars/uiKit/SocialList/SocialList';
import {socialLinks} from '@ecars/uiKit/SocialList/constants';

export const TopbarComponent: FC = () => (
  <div
    data-testid={TOPBAR_TEST_ID}
    className="topbar"
  >
    <Topbar.Wrapper>
      <div className="topbar__body">
        <SocialList
          items={socialLinks}
          color="lightgray"
        />
        <Topbar.Tools />
      </div>
    </Topbar.Wrapper>
  </div>
);
