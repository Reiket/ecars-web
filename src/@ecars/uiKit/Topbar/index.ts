import {TopbarWrapperHOC} from '@ecars/uiKit/Topbar/constants';
import {TopbarComponent} from '@ecars/uiKit/Topbar/TopbarComponent';
import {TopbarDropdown} from '@ecars/uiKit/Topbar/components/TopbarDropdown';
import {TopbarTools} from '@ecars/uiKit/Topbar/components/TopbarTools';
import {TopbarSocial} from '@ecars/uiKit/Topbar/components/TopbarSocial';

export const Topbar = Object.assign(TopbarComponent, {
  Wrapper: TopbarWrapperHOC,
  Dropdown: TopbarDropdown,
  Tools: TopbarTools,
  Social: TopbarSocial,
});
