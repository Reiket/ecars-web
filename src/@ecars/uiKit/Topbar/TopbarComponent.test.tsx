import {render, screen} from '@testing-library/react';
import {Topbar} from '@ecars/uiKit/Topbar';
import {TOPBAR_TEST_ID} from '@ecars/uiKit/Topbar/constants';

describe('Topbar Component', () => {
  test(`component renders correctly`, () => {
    const {container} = render(<Topbar />);
    const topbar = screen.getByTestId(TOPBAR_TEST_ID);
    expect(topbar).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
