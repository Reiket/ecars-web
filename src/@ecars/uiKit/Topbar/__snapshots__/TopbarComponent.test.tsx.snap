// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Topbar Component > component renders correctly 1`] = `
<div>
  <div
    class="topbar"
    data-testid="topbarTestId"
  >
    <section
      class="topbar__section section"
    >
      <div
        class="topbar__container container container--lg"
      >
        <div
          block="topbar"
          class="topbar__body"
        >
          <ul
            class="topbar__social social"
          >
            <li
              class="social__item"
            >
              <a
                class="topbar__link link link--lightgray link--icon"
                data-testid="routerLinkTestId"
                href="https://www.facebook.com"
                rel="noopener noreferrer"
                target="_blank"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 320 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M279.14 288l14.22-92.66h-88.91v-60.13c0-25.35 12.42-50.06 52.24-50.06h40.42V6.26S260.43 0 225.36 0c-73.22 0-121.08 44.38-121.08 124.72v70.62H22.89V288h81.39v224h100.17V288z"
                  />
                </svg>
              </a>
            </li>
            <li
              class="social__item"
            >
              <a
                class="topbar__link link link--lightgray link--icon"
                data-testid="routerLinkTestId"
                href="https://twitter.com"
                rel="noopener noreferrer"
                target="_blank"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 512 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"
                  />
                </svg>
              </a>
            </li>
            <li
              class="social__item"
            >
              <a
                class="topbar__link link link--lightgray link--icon"
                data-testid="routerLinkTestId"
                href="https://www.youtube.com"
                rel="noopener noreferrer"
                target="_blank"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 576 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M549.655 124.083c-6.281-23.65-24.787-42.276-48.284-48.597C458.781 64 288 64 288 64S117.22 64 74.629 75.486c-23.497 6.322-42.003 24.947-48.284 48.597-11.412 42.867-11.412 132.305-11.412 132.305s0 89.438 11.412 132.305c6.281 23.65 24.787 41.5 48.284 47.821C117.22 448 288 448 288 448s170.78 0 213.371-11.486c23.497-6.321 42.003-24.171 48.284-47.821 11.412-42.867 11.412-132.305 11.412-132.305s0-89.438-11.412-132.305zm-317.51 213.508V175.185l142.739 81.205-142.739 81.201z"
                  />
                </svg>
              </a>
            </li>
            <li
              class="social__item"
            >
              <a
                class="topbar__link link link--lightgray link--icon"
                data-testid="routerLinkTestId"
                href="https://www.instagram.com"
                rel="noopener noreferrer"
                target="_blank"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 1024 1024"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M512 378.7c-73.4 0-133.3 59.9-133.3 133.3S438.6 645.3 512 645.3 645.3 585.4 645.3 512 585.4 378.7 512 378.7zM911.8 512c0-55.2.5-109.9-2.6-165-3.1-64-17.7-120.8-64.5-167.6-46.9-46.9-103.6-61.4-167.6-64.5-55.2-3.1-109.9-2.6-165-2.6-55.2 0-109.9-.5-165 2.6-64 3.1-120.8 17.7-167.6 64.5C132.6 226.3 118.1 283 115 347c-3.1 55.2-2.6 109.9-2.6 165s-.5 109.9 2.6 165c3.1 64 17.7 120.8 64.5 167.6 46.9 46.9 103.6 61.4 167.6 64.5 55.2 3.1 109.9 2.6 165 2.6 55.2 0 109.9.5 165-2.6 64-3.1 120.8-17.7 167.6-64.5 46.9-46.9 61.4-103.6 64.5-167.6 3.2-55.1 2.6-109.8 2.6-165zM512 717.1c-113.5 0-205.1-91.6-205.1-205.1S398.5 306.9 512 306.9 717.1 398.5 717.1 512 625.5 717.1 512 717.1zm213.5-370.7c-26.5 0-47.9-21.4-47.9-47.9s21.4-47.9 47.9-47.9 47.9 21.4 47.9 47.9a47.84 47.84 0 0 1-47.9 47.9z"
                  />
                </svg>
              </a>
            </li>
            <li
              class="social__item"
            >
              <a
                class="topbar__link link link--lightgray link--icon"
                data-testid="routerLinkTestId"
                href="https://wa.me/971558119024"
                rel="noopener noreferrer"
                target="_blank"
              >
                <svg
                  data-testid="leftIconLinkTestId"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 512 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M260.062 32C138.605 32 40.134 129.701 40.134 250.232c0 41.23 11.532 79.79 31.559 112.687L32 480l121.764-38.682c31.508 17.285 67.745 27.146 106.298 27.146C381.535 468.464 480 370.749 480 250.232 480 129.701 381.535 32 260.062 32zm109.362 301.11c-5.174 12.827-28.574 24.533-38.899 25.072-10.314.547-10.608 7.994-66.84-16.434-56.225-24.434-90.052-83.844-92.719-87.67-2.669-3.812-21.78-31.047-20.749-58.455 1.038-27.413 16.047-40.346 21.404-45.725 5.351-5.387 11.486-6.352 15.232-6.413 4.428-.072 7.296-.132 10.573-.011 3.274.124 8.192-.685 12.45 10.639 4.256 11.323 14.443 39.153 15.746 41.989 1.302 2.839 2.108 6.126.102 9.771-2.012 3.653-3.042 5.935-5.961 9.083-2.935 3.148-6.174 7.042-8.792 9.449-2.92 2.665-5.97 5.572-2.9 11.269 3.068 5.693 13.653 24.356 29.779 39.736 20.725 19.771 38.598 26.329 44.098 29.317 5.515 3.004 8.806 2.67 12.226-.929 3.404-3.599 14.639-15.746 18.596-21.169 3.955-5.438 7.661-4.373 12.742-2.329 5.078 2.052 32.157 16.556 37.673 19.551 5.51 2.989 9.193 4.529 10.51 6.9 1.317 2.38.901 13.531-4.271 26.359z"
                  />
                </svg>
                +971 55 811 9024
              </a>
            </li>
            <li
              class="social__item"
            >
              <a
                class="topbar__link link link--lightgray link--icon"
                data-testid="routerLinkTestId"
                href="mailto:request@example.com"
                rel="noopener noreferrer"
                target="_blank"
              >
                <svg
                  data-testid="leftIconLinkTestId"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                  />
                  <path
                    d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4-8 5-8-5V6l8 5 8-5v2z"
                  />
                </svg>
                request@example.com
              </a>
            </li>
          </ul>
          <div
            class="topbar__tools"
          >
            <div
              class="topbar__dropdown-field dropdown-field"
            >
              <p
                class="dropdown-field__label"
              >
                Ship to:
              </p>
              <div
                class="topbar__dropdown-menu dropdown-menu dropdown-menu--dark"
                data-testid="dropdownTestId"
              >
                <div
                  class="dropdown-menu__block"
                >
                  <button
                    class="dropdown-menu__category"
                  >
                    <img
                      alt="ca"
                      src="./assets/images/flags/canada.svg"
                    />
                    Canada
                    <svg
                      class="dropdown-menu__arrow"
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 24 24"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M0 0h24v24H0V0z"
                        fill="none"
                      />
                      <path
                        d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
                      />
                    </svg>
                  </button>
                </div>
                <ul
                  class="dropdown-menu__list"
                >
                  <li
                    class="dropdown-menu__option _selected-option"
                    data-value="ca"
                  >
                    <img
                      alt="ca"
                      src="./assets/images/flags/canada.svg"
                    />
                     
                    Canada
                  </li>
                  <li
                    class="dropdown-menu__option"
                    data-value="us"
                  >
                    <img
                      alt="usa"
                      src="./assets/images/flags/usa.svg"
                    />
                     
                    USA
                  </li>
                  <li
                    class="dropdown-menu__option"
                    data-value="eu"
                  >
                    <img
                      alt="eu"
                      src="./assets/images/flags/europe.svg"
                    />
                     
                    Europe
                  </li>
                </ul>
              </div>
            </div>
            <div
              class="topbar__dropdown-field dropdown-field"
            >
              <p
                class="dropdown-field__label"
              >
                Currency:
              </p>
              <div
                class="topbar__dropdown-menu dropdown-menu dropdown-menu--dark"
                data-testid="dropdownTestId"
              >
                <div
                  class="dropdown-menu__block"
                >
                  <button
                    class="dropdown-menu__category"
                  >
                    US Dollar
                    <svg
                      class="dropdown-menu__arrow"
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 24 24"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M0 0h24v24H0V0z"
                        fill="none"
                      />
                      <path
                        d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
                      />
                    </svg>
                  </button>
                </div>
                <ul
                  class="dropdown-menu__list"
                >
                  <li
                    class="dropdown-menu__option _selected-option"
                    data-value="usd"
                  >
                     
                    US Dollar
                  </li>
                  <li
                    class="dropdown-menu__option"
                    data-value="eur"
                  >
                     
                    Euro
                  </li>
                  <li
                    class="dropdown-menu__option"
                    data-value="gbp"
                  >
                     
                    British Pound
                  </li>
                </ul>
              </div>
            </div>
          </div>
        </div>
      </div>
    </section>
  </div>
</div>
`;
